### PR TITLE
feat(fabricmc): init

### PR DESCRIPTION
**Pull request by KnownSH**
## 🎉 Theme for [FabricMC](https://fabricmc.net/) 🎉

preview (3)

## 🗒 Checklist 🗒
- [x] I have read and followed Catppuccin's [submission guidelines](https://github.com/catppuccin/userstyles/blob/main/docs/userstyle-creation.md).
- [x] I have made a new directory underneath `/styles/<name-of-website>` containing the contents of the [`/template`](https://github.com/catppuccin/userstyles/blob/main/template/) directory.
  - [x] I have ensured that the new directory is in **lower-kebab-case**.
  - [x] I have followed the template and kept the preprocessor as [LESS](https://lesscss.org/#overview).
- [x] I have made sure to update the
      [`userstyles.yml`](https://github.com/catppuccin/userstyles/blob/main/scripts/userstyles.yml)
      file with information about the new userstyle.
- [x] I have included the following files:
  - [x] `catppuccin.user.css` - all the CSS for the userstyle, based on the
        template.
  - [x] `preview.webp` - composite image of all four individual flavor screenshots (taken with the default accent color of mauve) stitched together, generated via [Catwalk](https://github.com/catppuccin/catwalk).

### CODE REVIEW

**Comment by KnownSH** [comment]
All three main parts of the website (docs.fabricmc.net **[VuePress]**, fabricmc.net, and fabricmc.net/wiki **[DokuWiki]**) should have proper styling. 
Note: VuePress uses [Prism](https://prismjs.com/) and DokuWiki uses [GeSHi](http://qbnz.com/highlighter/) for syntax highlighting.

**Comment by uncenter** [comment]
Looks like they use VitePress, not VuePress. We should hopefully have a theme we can import at some point instead of redefining all the variables here -  https://github.com/42Willow/vitepress. I also got around to finally building a Prism.js theme - https://github.com/uncenter/ctp-prismjs - so once that gets transferred we can include it here as well. I'll take a look at how we can theme GeSHi!

**Comment by uncenter** [comment]
Ah it's actually Shiki that's used, not Prism.js.

File: styles/fabricmc/catppuccin.user.css [hunk 1]
```diff
@@ -0,0 +1,695 @@
...
+
+    // Diff line removal
+    .gd {
+      background-color: fadeout(@red, 60%) !important;
+    }
+
+    // Diff line addition
+    .gi {
+      background-color: fadeout(@green, 60%) !important;
+    }
```
Comment (uncenter) [inline styles/fabricmc/catppuccin.user.css:221]: Should be 20% opacity at most as per https://github.com/catppuccin/catppuccin/blob/main/docs/style-guide.md#code-editors. Might want to group these lines with the other syntax highlighting as well.

File: styles/fabricmc/catppuccin.user.css [hunk 2]
```diff
@@ -0,0 +1,695 @@
...
+  }
+}
+@-moz-document domain('docs.fabricmc.net') {
+  :root.dark {
+    #catppuccin(@darkFlavor, @accentColor);
+  }
+
+  :root[class=""] {
+    #catppuccin(@lightFlavor, @accentColor);
+  }
```
Comment (uncenter) [inline styles/fabricmc/catppuccin.user.css:552]:
```suggestion
  :root.dark {
    #catppuccin(@darkFlavor, @accentColor);
  }

  :root:not(.dark) {
    #catppuccin(@lightFlavor, @accentColor);
  }
```

File: styles/fabricmc/catppuccin.user.css [hunk 3]
```diff
@@ -0,0 +1,695 @@
...
+      "<svg xmlns='http://www.w3.org/2000/svg' fill='none' stroke='@{surface2}' stroke-linecap='round' stroke-linejoin='round' stroke-width='2' viewBox='0 0 24 24'><rect width='8' height='4' x='8' y='2' rx='1' ry='1'/><path d='M16 4h2a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H6a2 2 0 0 1-2-2V6a2 2 0 0 1 2-2h2'/></svg>"
+    );
+    --vp-icon-copy: url("data:image/svg+xml,@{svg}");
+
+    @svg: escape(
+      "<svg xmlns='http://www.w3.org/2000/svg' fill='none' stroke='@{surface2}' stroke-linecap='round' stroke-linejoin='round' stroke-width='2' viewBox='0 0 24 24'><rect width='8' height='4' x='8' y='2' rx='1' ry='1'/><path d='M16 4h2a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H6a2 2 0 0 1-2-2V6a2 2 0 0 1 2-2h2'/><path d='m9 14 2 2 4-4'/></svg>"
+    );
+    --vp-icon-copied: url("data:image/svg+xml,@{svg}");
+
+    .banner[data-v-f9f361cc] {
```
Comment (uncenter) [inline styles/fabricmc/catppuccin.user.css:639]: Can we avoid using autogenerated attributes if possible?

**Review by uncenter — CHANGES_REQUESTED** [review]

File: styles/fabricmc/catppuccin.user.css (hunk 1 above)
Comment (uncenter) [reply styles/fabricmc/catppuccin.user.css:221]: Also these two need to have `color` set (to text or whatever), the default website styles reset it.

File: styles/fabricmc/catppuccin.user.css [hunk 4]
```diff
@@ -0,0 +1,695 @@
...
+      background-color: @base;
+      color: @text;
+    }
+
+    // <select> and <option> element styles are mostly used on fabricmc.net/develop
+    select,
+    input {
+      background-color: @mantle !important;
+      border-color: @surface1;
+    }
```
Comment (uncenter) [inline styles/fabricmc/catppuccin.user.css:118]:
```suggestion
    // <select> and <option> element styles are mostly used on fabricmc.net/develop
    select,
    input {
      background-color: @mantle !important;
      border-color: @surface1;
      color: @text;
    }
```

File: styles/fabricmc/catppuccin.user.css [hunk 5]
```diff
@@ -0,0 +1,695 @@
...
+        border-color: @surface2;
+      }
+
+      .pageId span {
+        color: @subtext0;
+      }
+
+      .page,
+      .aside {
+        .li {
```
Comment (uncenter) [inline styles/fabricmc/catppuccin.user.css:244]: Hmm why are list elements themed a different color? Doesn't look right on https://fabricmc.net/wiki/faq:user.

File: styles/fabricmc/catppuccin.user.css [hunk 6]
```diff
@@ -0,0 +1,695 @@
...
+      img[src="/wiki/lib/images/smileys/exclaim.svg"] {
+        @svg: escape(
+          '<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 36 36"><circle cx="18.001" cy="18.001" r="18" fill="@{yellow}"/><path fill="@{crust}" d="M15.578 26.993a2.423 2.423 0 1 1 4.846.004 2.423 2.423 0 0 1-4.846-.007zm.19-18.294c0-1.304.96-2.11 2.23-2.11 1.243 0 2.233.836 2.233 2.106v11.94c0 1.27-.99 2.11-2.23 2.11-1.274 0-2.234-.81-2.234-2.11z"/></svg>'
+        );
+        content: url("data:image/svg+xml,@{svg}");
+      }
+
+      .urlextern {
+        @svg: escape(
+          '<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path fill="@{surface2}" d="M17.9 17.39c-.26-.8-1.01-1.39-1.9-1.39h-1v-3a1 1 0 0 0-1-1H8v-2h2a1 1 0 0 0 1-1V7h2a2 2 0 0 0 2-2v-.41a7.984 7.984 0 0 1 2.9 12.8M11 19.93c-3.95-.49-7-3.85-7-7.93 0-.62.08-1.22.21-1.79L9 15v1a2 2 0 0 0 2 2m1-16A10 10 0 0 0 2 12a10 10 0 0 0 10 10 10 10 0 0 0 10-10A10 10 0 0 0 12 2"/></svg>'
```
Comment (uncenter) [inline styles/fabricmc/catppuccin.user.css:443]:
```suggestion
          '<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path fill="@{subtext0}" d="M17.9 17.39c-.26-.8-1.01-1.39-1.9-1.39h-1v-3a1 1 0 0 0-1-1H8v-2h2a1 1 0 0 0 1-1V7h2a2 2 0 0 0 2-2v-.41a7.984 7.984 0 0 1 2.9 12.8M11 19.93c-3.95-.49-7-3.85-7-7.93 0-.62.08-1.22.21-1.79L9 15v1a2 2 0 0 0 2 2m1-16A10 10 0 0 0 2 12a10 10 0 0 0 10 10 10 10 0 0 0 10-10A10 10 0 0 0 12 2"/></svg>'
```

**Review by uncenter — CHANGES_REQUESTED** [review]
Filename tabs unthemed on https://fabricmc.net/wiki/tutorial:datagen_setup.
Line numbers unthemed on https://fabricmc.net/wiki/tutorial:entity.

File: styles/fabricmc/catppuccin.user.css (hunk 1 above)
Comment (KnownSH) [reply styles/fabricmc/catppuccin.user.css]: > Might want to group these lines with the other syntax highlighting as well.
These are for the blog component of main website and are not used on the wiki (where the GeSHi ruleset is). I did change it so it is grouped them with other styles relating to the blogs syntax highlighting however.

File: styles/fabricmc/catppuccin.user.css [hunk 7]
```diff
@@ -0,0 +1,714 @@
...
+    }
+
+    // Diff line addition
+    .gi {
+      background-color: fadeout(@green, 15%) !important;
+    }
+
+    .gi, .gd {
+      color: @crust !important;
+    }
```
Comment (uncenter) [inline styles/fabricmc/catppuccin.user.css:147]:
```suggestion
    // Diff line removal
    .gd {
      background-color: fade(@red, 15%) !important;
    }

    // Diff line addition
    .gi {
      background-color: fade(@green, 15%) !important;
    }

    .gi, .gd {
      color: @text !important;
    }
```

**Review by uncenter — CHANGES_REQUESTED** [review]
The Javadoc select on https://fabricmc.net/develop/ has unthemed text.

File tab at bottom of https://fabricmc.net/wiki/tutorial:datagen_setup unthemed.

Advanced concepts table at https://fabricmc.net/wiki/tutorial:commands unthemed.

**Comment by KnownSH** [comment]
Unfortunately, I won't be able to complete this pr due to a combined lack of motivation and preoccupation with work.

**Pull request closed without merging**